releases/release-4.0.10.md

```diff
@@ -0,0 +1,68 @@
+---
+title: TiDB 4.0.10 Release Notes
+---
+
+# TiDB 4.0.10 Release Notes
+
+Release date: January 15, 2021
+
+TiDB version: 4.0.10
+
+## Compatibility Changes
+
+## New Features
+
+## Improvements
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix a concurrency bug that may cause the batch client timeout [#22336](https://github.com/pingcap/tidb/pull/22336)
+    - Avoid duplicate bindings caused by concurrent baseline capture [#22295](https://github.com/pingcap/tidb/pull/22295)
+    - Make baseline capture work when log level is 'debug' [#22293](https://github.com/pingcap/tidb/pull/22293)
+    - Fix gc miss locks when region merged during scanning & resolving locks [#22267](https://github.com/pingcap/tidb/pull/22267)
+    - Return correct results for user variables of datetime type [#22143](https://github.com/pingcap/tidb/pull/22143)
+    - Avoid using index merge when there are multiple table filters. [#22124](https://github.com/pingcap/tidb/pull/22124)
+    - Fixed the issue that using prepare statement to cache execution plan caused tiflash to return `Wrong precision`. [#21960](https://github.com/pingcap/tidb/pull/21960)
+    - Fix a bug that schema change will lead to incorrectness when forUpdateRead [#21596](https://github.com/pingcap/tidb/pull/21596)
+    - Fix a bug that doesn't add NoDefaultValueFlag when creating a table [#21474](https://github.com/pingcap/tidb/pull/21474)
+    - Set dbName for hinted query block table alias [#21380](https://github.com/pingcap/tidb/pull/21380)
+    - Generate proper hint for IndexHashJoin / IndexMergeJoin [#21020](https://github.com/pingcap/tidb/pull/21020)
+
++ TiFlash
+
+    - Fix the problem that TiFlash fail to start when path_realtime_mode is true [#1322](https://github.com/pingcap/tics/pull/1322)
+    - Fixed an issue where tables created in the old version of tidb may cause the startup of the new tiflash node to fail. [#1297](https://github.com/pingcap/tics/pull/1297)
```

Comment:
```suggestion
    - Fixed an issue that TiFlash can't start because TiFlash failed to process the schema of TiDB with an old version.
    - Fixed an issue where TiFlash can't start due to incorrect handling of cpu_time on the RedHat system.
    - Fixed the issue that TiFlash fails to start when path_realtime_mode is true.
    - Fixed the issue of incorrect results when calling the `substr` function with three parameters.
```